Azure Services/Azure Monitor/Agents/Migration Tools/DCR Config Generator/VMIDcrGeneratorModule/VMIDcrGeneratorModule.psm1

```diff
@@ -0,0 +1,194 @@
+       $True
+    }
+    
+    return $False
+}
+
+function Get-VmiDataSources
+{
+    param (
+        [Parameter(Mandatory=$true)][bool] $ProcessAndDependencies
+    )
+    
+    $vmiPerfCounter = [ordered]@{
+        "name" = "VMInsightsPerfCounters";
+        "streams" = @("Microsoft-InsightsMetrics");
+        "samplingFrequencyInSeconds" = 60;
+        "counterSpecifiers" = @("\VmInsights\DetailedMetrics");
+    }
+    $vmiPerfCounters = @($vmiPerfCounter)
+    $vmiDataSources = [ordered]@{
+        "performanceCounters" = $vmiPerfCounters;
+    }
+    if ($ProcessAndDependencies) {
+        $vmiExtension = [ordered]@{
+        "streams" = @("Microsoft-ServiceMap");
+        "extensionName" = "DependencyAgent";
+        "extensionSettings" = @{};
+        "name" = "DependencyAgentDataSource";
+        }
+        $vmiDataSources["extensions"] = @($vmiExtension)
+    }    
+    return $vmiDataSources
+}
+
+function Get-VmiDestinations
+{
+    param (
+        [Parameter(Mandatory=$true)][string] $ResourceGroupName,
+        [Parameter(Mandatory=$true)][string] $WorkspaceName
+    )
+    
+    $workspace = Get-AzOperationalInsightsWorkspace -ResourceGroupName $ResourceGroupName -Name $WorkspaceName
+    $laDest =
+    [ordered]@{
+        "workspaceResourceId" = $workspace.ResourceId;
+        "workspaceId" = $workspace.CustomerId; 
+        "name" = "VMInsightsPerf-Logs-Dest";
+    }
+    $vmiDestinations = 
+    @{
+        "logAnalytics" = @($laDest);
+    }
+    return $vmiDestinations
+}
+
+function Get-VmiDataFlows
+{
+    param (
+        [Parameter(Mandatory=$true)][bool] $ProcessAndDependencies
+    )
+
+    $vmiDataFlows = [System.Collections.ArrayList]::new()
+    $insightDataFlow =
+    [ordered]@{
+        "streams" = @("Microsoft-InsightsMetrics");
+        "destinations" = @("VMInsightsPerf-Logs-Dest");
+    }
+    $serviceMapDataFlow =
+    [ordered]@{
+        "streams" = @("Microsoft-ServiceMap");
+        "destinations" = @("VMInsightsPerf-Logs-Dest");
+    }
+    $vmiDataFlows.Add($insightDataFlow) | Out-Null
+    if ($ProcessAndDependencies) {
+        $vmiDataFlows.Add($serviceMapDataFlow) | Out-Null
+    }
+    return $vmiDataFlows
+}
+
+function Get-VmiDcrPayload
+{
+  param (
+        [Parameter(Mandatory=$true)][bool] $ProcessAndDependencies,
+        [Parameter(Mandatory=$true)][string] $ResourceGroupName,
+        [Parameter(Mandatory=$true)][string] $WorkspaceName
+    )
+
+    $vmiDataSources = Get-VmiDataSources -ProcessAndDependencies $ProcessAndDependencies
+    $vmiDataDestinations = Get-VmiDestinations -ResourceGroupName $ResourceGroupName -WorkspaceName $WorkspaceName
+    $vmiDataFlows = Get-VmiDataFlows -ProcessAndDependencies $ProcessAndDependencies
+    $vmiDcrPayload = [ordered]@{
+        "properties" = [ordered]@{
+                "dataSources" = $vmiDataSources;
+                "destinations" = $vmiDataDestinations;
+                "dataFlows" = @($vmiDataFlows);

```

Comment:
Cosmetic change, if a DCR only collect InsighsMetrics -> the dataflows is not enclosed in a array and rather just a single item, this is not again have a functional impact, but writing this way the DCR will be inline with the DCRs are produced in the portal or available to download as DCR ARM Templates.